File: 280 Wiggle Sort.py
'''
Wiggle Sort
https://leetcode.com/problems/wiggle-sort/
'''
def wiggleSort(nums):
    '''
    if i is oddy, nums[i-1] <= nums[i]
    if i is even, nums[i-1] >= nums[i]
    '''
    for i in xrange(1,len(nums)):
        if ((i & 1) == 1 and nums[i-1] > nums[i]) or ((i & 1) == 0 and nums[i-1] < nums[i]):
            nums[i-1], nums[i] = nums[i], nums[i-1]

nums =  [3, 5, 2, 1, 6, 4]
wiggleSort(nums)
print nums           

File: 280 Wiggle Sort.cpp
/*
280. Wiggle Sort
https://leetcode.com/problems/wiggle-sort/
    
Given an unsorted array nums, reorder it in-place such that nums[0] <= nums[1] >= nums[2] <= nums[3]....

Example:

Input: nums = [3,5,2,1,6,4]
Output: One possible answer is [3,5,1,6,2,4]
*/
#import <algorithm>
class Solution
{
public:
    void wiggleSort(vector<int>& nums)
    {
        if (nums.size() == 0)
        {
            return;    
        }
        
        bool oddy = true;
        for(int i = 0; i < nums.size()-1; i++)
        {
            if (oddy)
            {
                if(nums[i] > nums[i+1])
                {
                    std::swap(nums[i], nums[i+1]);
                }
            }
            else
            {
                if(nums[i] < nums[i+1])
                {
                    std::swap(nums[i], nums[i+1]);
                }
            }                        
            oddy = !oddy;
        }      
    }
};
